File: prob14.js
// https://projecteuler.net/problem=14

let newNum = 0;;
let largestChain = 0;
let largestNum = 0;

for(let current = 1; current < 1000000; current++){
  let chainLength = 0;
  newNum = current;
  while(newNum !== 1){
    if(newNum % 2 === 0){
      newNum = newNum / 2;
      chainLength++;
    }else{
      newNum = newNum * 3 + 1
      chainLength++;
    }
  }
  if(chainLength > largestChain){
    largestNum = current;
    largestChain = chainLength;
  }
}

console.log(largestNum);

File: prob14.py
# https://projecteuler.net/problem=14
# Completed 1/4/2022

largest = 0
largestCount = 0

for i in range(1, 1_000_000):
    sequenceNum = i
    count = 0
    while sequenceNum != 1:
        if sequenceNum % 2 == 0:
            sequenceNum /= 2
        else:
            sequenceNum = sequenceNum * 3 + 1
        count += 1
    if count > largestCount:
        largestCount = count
        largest = i
        
print(largest)